cpp/src/io/orc/reader_impl.cu

```diff
@@ -1045,6 +1045,7 @@ table_with_metadata reader::impl::read(size_type skip_rows,
           auto dst_base = static_cast<uint8_t*>(stripe_data.back().data());
 
           // Coalesce consecutive streams into one read
+          std::vector<std::pair<std::future<size_t>, size_t>> read_tasks;
```

Comment:
It would be better to move this out of the loop, so we don't sync for each stripe.
CC @rgsl888prabhu

Reply:
Actually, you have a similar situation in the Parquet reader, too. Would it be possible to increase the scope of the futures to only sync once?

Reply:
Modest speedups in parquet: 117 ms -> 104 ms. https://github.com/rapidsai/cudf/pull/8752/commits/7af4c6a789232387049f47c0ff2f4358e6e139f3

Reply:
Modest speedup in parquet 117ms -> 104ms https://github.com/rapidsai/cudf/pull/8752/commits/7af4c6a789232387049f47c0ff2f4358e6e139f3

Reply:
Similar absolute speedup in ORC. 97 ms -> 83 ms https://github.com/rapidsai/cudf/pull/8752/commits/73a22ec7365772c556a47446747404bf9ac5be09. Now no kernels can be launched in this loop that depend on the data having been read.